Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#include "Components/PSHUDComponent.h"
+//---
+#include "Bomber.h"
+#include "Controllers/MyPlayerController.h"
+#include "UtilityLibraries/MyBlueprintFunctionLibrary.h"
+#include "Data/PSDataAsset.h"
+#include "Blueprint/WidgetTree.h"
+#include "UI/MyHUD.h"
+#include "Widgets/PSMenuWidget.h"
+#include "Data/PSTypes.h"
+#include "Data/PSSaveGameData.h"
+#include "Data/PSWorldSubsystem.h"
+#include "Components/Image.h"
+#include "GameFramework/MyGameStateBase.h"
+#include "GameFramework/MyPlayerState.h"
+#include "MyDataTable/MyDataTable.h"
+#include "MyUtilsLibraries/WidgetUtilsLibrary.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(PSHUDComponent)
+
+// Sets default values for this component's properties
+UPSHUDComponent::UPSHUDComponent()
+{
+	// Set this component to be initialized when the game starts, and to be ticked every frame.  You can turn these features
+	// off to improve performance if you don't need them.
+	PrimaryComponentTick.bCanEverTick = false;
+	PrimaryComponentTick.bStartWithTickEnabled = false;
+}
+
+// Called when the game starts
+void UPSHUDComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	AMyHUD* MyHUD = Cast<AMyHUD>(GetOwner());
+	const AMyHUD& HUD = *MyHUD;
```

Comment:
`*MyHUD` could cause silent crash. Use `CastChecked` instead of `Cast` on the line above, so it will crash the same, but not silently, with extra message if failed to cast to given class

Reply:
improved cast to `CastChecked`. For me sounds logically always to use `CastChecked`. 
So the lesson here: always use `CastChecked`? Is there a case when to use only `Cast`?

Reply:
| Is there a case when to use only Cast?
Unlike c++ static_cast\dynamic_cast, both Unreal's `Cast` and `CastChecked` before casting also do regular `!= nullptr` check.  **It's the main point**.
So in many cases any Cast could fail just because original object was already destroyed. So it allows developers avoid doing unnecessary null checks before casting.
Before of that main point, `CastChecked` can not be always used, it will crash your game the same as `checkf` does. So we use CastChecked in the same cases when `checkf`, when its FATAL error and we don't allow user to play such broken game (like if Data Asset itself is not loaded, so progression system if completely broken).
However, we shouldn't use 'CastChecked' where there is any chance (even very small) of any gameplay bug, where gamer still should be able to play